### DISCUSSION TITLE
Running Server Actions causes: Node.js process exited with signal: 11 (SIGSEGV) (core dumped).

### DISCUSSION THREAD

**Discussion by ImJustChew** [vercel/next.js#67638] [Help]
### Summary

We've recently migrated all our API routes to Server Actions. However, when we try to execute functions now, it randomly crashes the server on Vercel, with a very unhelpful log

`Node.js process exited with signal: 11 (SIGSEGV) (core dumped). The logs above can help with debugging the issue.`

![image](https://github.com/vercel/next.js/assets/17233431/b3d4ad98-ea5c-48fb-8333-894592458b5f)

The code runs fine on the dev server, but fails on Vercel. Has anyone had similar experiences? or is there any way to view more helpful logs on vercel to debug this issue?

**Comment by icyJoseph** [comment]
What kind of work is this server action doing? Is the same version of Node.js setup locally, and in Vercel?

**Comment by ImJustChew** [comment]
I've checked that the Node.js Version is the same, v20. These server
actions fetch student data from the University, handle authentication etc.
And It only started after we migrated more functions to server actions.

Our repo is https://github.com/nthumodifications/courseweb/issues

**Reply by pacificnm** [reply]
Following. Seeing this same issue, same version of node. Using prisma and Authjs. Will provide more details in the morning

**Comment by ImJustChew** [comment]
Vercel is also heavily truncating Nextjs logs, thus, we cannot debug exactly what and where is causing these crashes. For now, we've moved our operations to DigitalOcean and everything is working as it should.

**Comment by limoragni** [comment]
Getting the same error on my end on Vercel. Very difficult to debug or understand what's happening.

**Reply by icyJoseph** [reply]
Ahh this issue 🤔 

Is there any chance to look at some code? or understand what the server actions are doing? 

Next.js version, Node?

**Reply by limoragni** [reply]
I see a few other erros before that one happening

double free or corruption (out)
malloc(): unsorted double linked list corrupted

and this is a code path that normally works well, it's an edpoint that calls an external API, and does a bunch of reads using Prisma. And there's some computation, I create a few Map in memory for lookups. 

I am using fluid compute, and those happen in different request IDs in Vercel. since it's happening on the same instance.

The Node version is 22. My guess this maybe an issue with Prisma. But very difficult to debug.

**Reply by limoragni** [reply]
Might be related to this

https://github.com/prisma/prisma/issues/27785

**Reply by rafalzawadzki** [reply]
also tracked here: https://community.vercel.com/t/random-http-500-errors-with-node-process-crashes-sigabrt-sigsegv/24249/12
